### PR TITLE
Update kubectl help for 1.2 resources

### PR DESCRIPTION
**Pull request by janetkuo**
Ref #23231; follow up #23222 
1. add deployment, rs, daemonsets, configmaps, and componentstatuses to the supported types 
2. update rs as supported in `kubectl autoscale` and `kubectl scale`
3. update `kubectl expose` and `kubectl rollout` help docs; add more examples

@bgrant0607 @kubernetes/kubectl @nikhiljindal @madhusudancs

### CODE REVIEW

File: pkg/kubectl/kubectl.go [hunk 1]
```diff
@@ -28,7 +28,15 @@ import (
...
 
-const kubectlAnnotationPrefix = "kubectl.kubernetes.io/"
+const (
+	kubectlAnnotationPrefix = "kubectl.kubernetes.io/"
+	// TODO: auto-generate this
+	PossibleResourceTypes = `Possible resource types include (case insensitive): pods (po), services (svc), deployments,
+replicasets (rs), replicationcontrollers (rc), nodes (no), events (ev), limitranges (limits),
+persistentvolumes (pv), persistentvolumeclaims (pvc), resourcequotas (quota), namespaces (ns),
+serviceaccounts, ingresses (ing), horizontalpodautoscalers (hpa), daemonsets (ds), configmaps,
+componentstatuses (cs), endpoints (ep) or secrets.`
```
Comment (madhusudancs) [inline pkg/kubectl/kubectl.go]: s/or/and/

File: pkg/kubectl/cmd/rollout/rollout.go [hunk 2]
```diff
@@ -24,7 +24,7 @@ import (
 )
 
 const (
-	rollout_long    = `rollout manages a deployment using subcommands like "kubectl rollout undo deployment/abc"`
+	rollout_long    = `kubectl rollout manages a deployment using subcommands like "kubectl rollout undo deployment/abc"`
```
Comment (madhusudancs) [inline pkg/kubectl/cmd/rollout/rollout.go]: "kubectl rollout" isn't necessary, you can just start with "Manages a deployment ..."? It looks like that's what we are doing with other commands.

File: pkg/kubectl/kubectl.go (hunk 1 above)
Comment (janetkuo) [inline pkg/kubectl/kubectl.go]: Fixed

File: pkg/kubectl/cmd/rollout/rollout.go (hunk 2 above)
Comment (janetkuo) [inline pkg/kubectl/cmd/rollout/rollout.go]: Done

**Comment by janetkuo** [comment]
Comments addressed.

File: pkg/kubectl/cmd/autoscale.go [hunk 3]
```diff
@@ -31,7 +31,7 @@ import (
 const (
 	autoscaleLong = `Creates an autoscaler that automatically chooses and sets the number of pods that run in a kubernetes cluster.
 
-Looks up a deployment or replication controller by name and creates an autoscaler that uses this deployment or replication controller as a reference.
+Looks up a Deployment, ReplicaSet or ReplicationController by name and creates an autoscaler that uses the given resource as a reference.
```
Comment (jlowdermilk) [inline pkg/kubectl/cmd/autoscale.go]: Nit: ReplicaSet,

File: pkg/kubectl/cmd/expose.go [hunk 4]
```diff
@@ -37,14 +37,14 @@ type ExposeOptions struct {
...
 const (
-	expose_long = `Take a replication controller, service, replica set, deployment or pod and expose it as a new Kubernetes service.
-
-Looks up a replication controller, service, replica set, deployment or pod by name and uses the selector 
-for that resource as the selector for a new service on the specified port. A replica set will be exposed 
-as a service only if it's selector is convertible to a selector that service supports, i.e. when the
-replica set selector contains only the matchLabels component. Note that if no port is specified
-via --port and the exposed resource has multiple ports, all will be re-used by the new service. Also
-if no labels are specified, the new service will re-use the labels from the resource it exposes.`
+	expose_long = `Take a deployment, service, replica set, replication controller or pod and expose it as a new Kubernetes service.
```
Comment (jlowdermilk) [inline pkg/kubectl/cmd/expose.go]: Nit: replication controller,

File: pkg/kubectl/cmd/expose.go [hunk 5]
```diff
@@ -37,14 +37,14 @@ type ExposeOptions struct {
...
+
+Looks up a deployment, service, replica set, replication controller or pod by name and uses the selector
+for that resource as the selector for a new service on the specified port. A deployment or replica set
+will be exposed as a service only if it's selector is convertible to a selector that service supports,
```
Comment (jlowdermilk) [inline pkg/kubectl/cmd/expose.go]: s/it's/its

File: pkg/kubectl/cmd/scale.go [hunk 6]
```diff
@@ -37,14 +37,14 @@ type ScaleOptions struct {
 }
 
 const (
-	scale_long = `Set a new size for a Replication Controller, Job, or Deployment.
+	scale_long = `Set a new size for a Deployment, ReplicaSet, Replication Controller or Job.
```
Comment (jlowdermilk) [inline pkg/kubectl/cmd/scale.go]: nit: Oxford comma here and below

File: pkg/kubectl/cmd/autoscale.go (hunk 3 above)
Comment (janetkuo) [inline pkg/kubectl/cmd/autoscale.go]: Fixed

File: pkg/kubectl/cmd/expose.go (hunk 4 above)
Comment (janetkuo) [inline pkg/kubectl/cmd/expose.go]: Fixed

File: pkg/kubectl/cmd/expose.go (hunk 5 above)
Comment (janetkuo) [inline pkg/kubectl/cmd/expose.go]: Fixed

File: pkg/kubectl/cmd/scale.go (hunk 6 above)
Comment (janetkuo) [inline pkg/kubectl/cmd/scale.go]: Fixed

**Comment by janetkuo** [comment]
@jlowdermilk All comments addressed. PTAL

**Comment by jlowdermilk** [comment]
lgtm. Feel free to add label after squashing commits.

**Comment by janetkuo** [comment]
Squashed and applying tag.

**Comment by janetkuo** [comment]
Travis failed to do its job. Reopening the PR. 

```
Skipping hack/../hack/verify-all.sh
Verifying hack/../hack/verify-api-reference-docs.sh
Generating api reference docs at /home/travis/gopath/src/github.com/kubernetes/kubernetes/_output/generated_html
Reading swagger spec from: /home/travis/gopath/src/github.com/kubernetes/kubernetes/api/swagger-spec/
Unable to find image 'gcr.io/google_containers/gen-swagger-docs:v5' locally
v5: Pulling from google_containers/gen-swagger-docs
No output has been received in the last 10m0s, this potentially indicates a stalled build or something wrong with the build itself.
The build has been terminated
```

**Comment by janetkuo** [comment]
@k8s-bot e2e test this issue: #23321

**Pull request merged**